.github/workflows/clang-format-pull-request.yml

```diff
@@ -0,0 +1,34 @@
+name: clang-format-pull-request
+
+on:
+  pull_request:
+    branches:
+      - release_oneDPL
+      - master
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 1
+
+    - name: Get clang-format
+      run: sudo apt-get install -yqq clang-format-6.0
+
+    - name: Applying clang-format for changed files
+      run: |
+        URL="https://api.github.com/repos/${GITHUB_REPOSITORY}/pulls/${{ github.event.pull_request.number }}/files"
+        FILES=$(curl -s --request GET --get $URL |  jq -r '.[] | .filename')
+        echo $FILES | xargs -n1 -t clang-format --style=file -i
```

Comment:
I'd suggest to run clang-format on all files in the repo. It's not consume too much time and won't affect files with the correct format since we assume the `clang-format` is already applied for all files in the repo.

Reply:
Many files in current source base does not conform formatting rules, so It was agreed to check only files changed.

Reply:
The check will run for all PR commit.  Scan all file will show irrelevant failures which will confuse contributors.